docs/ci-services.md

```diff
@@ -0,0 +1,102 @@
+# CI Services
+
+The CI services map defines the supported CI services the user will be
+presented with when running the scaffolder.
+
+## Example
+
+```javascript
+{
+  CIServiceName: {scaffolder: opts => scaffold(opts), public: true}
+}
+```
+
+## Existing scaffolders
+There are already multiple CI service scaffolders [available for
+consumption](https://github.com/form8ion/awesome#continuous-integration-services).
+Should your workflow fit into those, feel free to integrate them directly. If
+not, you are welcome and encouraged to build more (and maybe consider
+submitting a PR with a link 😎)!
+
+## Config
+
+### keys: __string__
+
+Keys in the CI services map define the choices the user will be presented with
+when choosing their desired CI service. These have no functional impact on the
+scaffolder.
+
+example: `CIServiceName`
+
+### values: __object__
+
+Values in the CI services map define the actual mechanisms and configuration
+that will be used by the JavaScript Scaffolder to scaffold the CI service for
+the user's project.
+
+#### scaffolder: __function__ (_required_)
+
+Consumer-provided plugin called when this CI service is selected by the
+user in the interactive prompt. Will be called with the following arguments:
+
+* `projectRoot` __string__ full path to the directory where scaffolder will
+  create the JavaScript project
+* `vcs` __object__
+  * `host` __string__ VCS hosting service to generate the repo with (e.g., `github`)
```

Comment:
one of the areas that i think we need to figure out a better way to use a constant throughout, but i think `github` is currently passed as `GitHub` everywhere that i can think of. did you find a situation where it is all lowercase?

Reply:
good call. i thought i had found all of these somewhere being used, but this one all i can find is `GitHub` like you suggested so i'm not sure how i ended there before. updating.